presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -348,12 +349,13 @@ public static String getDeserializerClassName(Properties schema)
     }
 
     @SuppressWarnings("deprecation")
-    public static Deserializer getDeserializer(Properties schema)
+    public static Deserializer getDeserializer(Configuration configuration, Properties schema)
     {
         String name = getDeserializerClassName(schema);
 
         Deserializer deserializer = createDeserializer(getDeserializerClass(name));
-        initializeDeserializer(deserializer, schema);
+        configuration = copy(configuration); // Some SerDes (e.g. Avro) modify passed configuration
```

Comment:
move to `initializeDeserializer`

Reply:
why?

Reply:
because `initializeDeserializer` is the place when we leave "our code" and call the unknown